# Cleveland Clinic Heart Disease Dataset Audit

## Objective
- Understanding the structure of dataset

## Import Library and load data

In [14]:
import pandas as pd

df = pd.read_csv("../data/raw/cleveland.csv")

df.head()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,num
0,63,1,1,145,233,1,2,150,0,2.3,3,0,6,0
1,67,1,4,160,286,0,2,108,1,1.5,2,3,3,2
2,67,1,4,120,229,0,2,129,1,2.6,2,2,7,1
3,37,1,3,130,250,0,0,187,0,3.5,3,0,3,0
4,41,0,2,130,204,0,2,172,0,1.4,1,0,3,0


## Dataset Shape

In [15]:
rows,cols = df.shape

print(f"Rows : {rows}")
print(f"Columns : {cols}")

Rows : 303
Columns : 14


## Columns

In [16]:
df.columns.tolist()

['age',
 'sex',
 'cp',
 'trestbps',
 'chol',
 'fbs',
 'restecg',
 'thalach',
 'exang',
 'oldpeak',
 'slope',
 'ca',
 'thal',
 'num']

## Data Types

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trestbps  303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalach   303 non-null    int64  
 8   exang     303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    int64  
 11  ca        303 non-null    object 
 12  thal      303 non-null    object 
 13  num       303 non-null    int64  
dtypes: float64(1), int64(11), object(2)
memory usage: 33.3+ KB


## Missing Values

In [18]:
df.isna().sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
num         0
dtype: int64

## Duplicate Values

In [19]:
df.duplicated().sum()

0

## Basic Statistics

### Numerical

In [20]:
df.describe()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,num
count,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000
mean,54.438944,0.679868,3.158416,131.689769,246.693069,0.148515,0.990099,149.607261,0.326733,1.039604,1.600660,0.937294
std,9.038662,0.467299,0.960126,17.599748,51.776918,0.356198,0.994971,22.875003,0.469794,1.161075,0.616226,1.228536
min,29.000000,0.000000,1.000000,94.000000,126.000000,0.000000,0.000000,71.000000,0.000000,0.000000,1.000000,0.000000
25%,48.000000,0.000000,3.000000,120.000000,211.000000,0.000000,0.000000,133.500000,0.000000,0.000000,1.000000,0.000000
50%,56.000000,1.000000,3.000000,130.000000,241.000000,0.000000,1.000000,153.000000,0.000000,0.800000,2.000000,0.000000
75%,61.000000,1.000000,4.000000,140.000000,275.000000,0.000000,2.000000,166.000000,1.000000,1.600000,2.000000,2.000000
max,77.000000,1.000000,4.000000,200.000000,564.000000,1.000000,2.000000,202.000000,1.000000,6.200000,3.000000,4.000000


### Categorical

In [21]:
df.describe(include="object")

,ca,thal
count,303,303
unique,5,4
top,0,3
freq,176,166


## Target Distribuition

In [22]:
df["num"].value_counts()

num
0    164
1     55
2     36
3     35
4     13
Name: count, dtype: int64

## Unique Values

In [23]:
df.nunique()

age          41
sex           2
cp            4
trestbps     50
chol        152
fbs           2
restecg       3
thalach      91
exang         2
oldpeak      40
slope         3
ca            5
thal          4
num           5
dtype: int64

## Dataset Dictionary
| Column   | Meaning                                                                                                        | Type                  |
| -------- | -------------------------------------------------------------------------------------------------------------- | --------------------- |
| age      | Age of the patient (years)                                                                                     | Numeric               |
| sex      | Sex of the patient (1 = Male, 0 = Female)                                                                      | Binary                |
| cp       | Chest pain type (1 = Typical angina, 2 = Atypical angina, 3 = Non-anginal pain, 4 = Asymptomatic)              | Categorical           |
| trestbps | Resting blood pressure (mm Hg on admission)                                                                    | Numeric               |
| chol     | Serum cholesterol (mg/dL)                                                                                      | Numeric               |
| fbs      | Fasting blood sugar > 120 mg/dL (1 = True, 0 = False)                                                          | Binary                |
| restecg  | Resting electrocardiographic results (0 = Normal, 1 = ST-T wave abnormality, 2 = Left ventricular hypertrophy) | Categorical           |
| thalach  | Maximum heart rate achieved                                                                                    | Numeric               |
| exang    | Exercise-induced angina (1 = Yes, 0 = No)                                                                      | Binary                |
| oldpeak  | ST depression induced by exercise relative to rest                                                             | Numeric               |
| slope    | Slope of the peak exercise ST segment (1 = Upsloping, 2 = Flat, 3 = Downsloping)                               | Categorical (Ordinal) |
| ca       | Number of major vessels (0–3) colored by fluoroscopy                                                           | Numeric (Discrete)    |
| thal     | Nuclear stress test result (3 = Normal, 6 = Fixed defect, 7 = Reversible defect)                               | Categorical           |
| num      | Heart disease diagnosis (0 = No disease, 1–4 = Increasing severity of heart disease)                           | Multiclass Target     |


## Initial Observation
- Dataset has 303 rows and 14 columns
- Dataset has both numerical and categorical Value
- Dataset has no missing and duplicate values
- num is our target variable
- Numerical features
    - age
    - trestbps
    - chol
    - thalach
    - exang
    - oldpeak
    - slope
- Binary features
    - sex
    - fbs
    - exang
- Categorical features
    - cp
    - restecg
    - slope
    - ca
    - thal
- Target is not balanced
- We only want to do binary classification so we have to process target variable
- Also "ca" and "thal" is numerical not object
    - due to cleveland standard to represent missing value as "?"